src/ipa_pc/mod.rs

```diff
@@ -300,7 +307,11 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
             }
         }
 
-        assert_eq!(proof.hiding_comm.is_some(), proof.rand.is_some());
+        // TODO: check error message
```

Comment:
"Hiding commitment and proof randomness must be both either present or not present"